src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,94 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getTodos, getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
 
-export const TodoModal: React.FC = () => {
+type TodoModalProps = {
+  todoId?: number;
```

Comment:
now, we will pass the selectedTodo here instead of the Id
```suggestion
  todo: Todo;
```
and this prop will be required because we render this only when that todo exists